src/storage/mvcc/reader/util.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use storage::mvcc::{Error, Result};
+use storage::mvcc::{Lock, LockType, Write};
+use storage::{Cursor, Iterator, Key, Statistics, Value};
+
+/// Representing check lock result.
+#[derive(Debug)]
+pub enum CheckLockResult {
+    /// Key is locked. The key lock error is included.
+    Locked(Error),
+
+    /// Key is not locked.
+    NotLocked,
+
+    /// Key's lock exists but was ignored because of requesting the latest committed version
+    /// for the primary key. The committed version is included.
+    Ignored(u64),
+}
+
+/// Load the lock from current cursor and check lock.
+/// This function requires that the cursor is pointing at the lock of the user key.
+#[inline]
+pub fn load_and_check_lock_from_cursor<I>(
+    lock_cursor: &mut Cursor<I>,
+    user_key: &Key,
+    ts: u64,
+    statistics: &mut Statistics,
+) -> Result<CheckLockResult>
+where
+    I: Iterator,
+{
+    assert!(lock_cursor.valid());
```

Comment:
the `key()` and `value()` of DBIterator use `assert!(self.valid())` innerly, seems that no need to call it here.

Reply:
You are right! I removed it. PTAL